### PR TITLE
Various cosmetic improvements

### PR DESCRIPTION
**Pull request by alexreg**
Related to the larger effort of https://github.com/rust-lang/rust/pull/58036.

### CODE REVIEW

**Comment by oli-obk** [comment]
reviewing these PRs takes a lot of time. The one on miri took me around an hour for the first round (it was slightly bigger to be fair).

I will not be making this one a priority, I have little enough time for clippy as it is. If noone else has the time and takes this PR up for review in the next two weeks, I will close it to keep our queue free.

**Comment by phansch** [comment]
Thanks, this looks like a welcome change in general!

As Oli mentioned, reviewing big PRs takes time, which is something we don't have as much of in Clippy (AFAIK no one is getting paid to work on Clippy currently?) and as it stands currently, we would prioritize bug-fixes and new lints over this PR.

To proceed, I would suggest to split this PR up into smaller ones to make it more easy for us to merge these changes. (i.e., re-ordering `use` statements, changing things that change lots of UI test output, documentation improvements, etc)

I'm going to leave this open in case someone else wants to take on the review of this PR. However, please don't hesitate to open new smaller PRs in the meantime.

**Comment by flip1995** [comment]
Since I'm actually paid as a research assistant, I will pick this up and review it. :+1:

File: tests/ui/len_zero.rs [hunk 1]
```diff
@@ -63,7 +63,8 @@ impl PubTraitsToo for One {
 }
 
 trait TraitsToo {
-    fn len(self: &Self) -> isize; // no error, len is private, see #1085
+    fn len(self: &Self) -> isize;
+    // No error; `len` is private; see issue #1085.
```
Comment (flip1995) [inline tests/ui/len_zero.rs]: These changes require a stderr update.

**Comment by flip1995** [comment]
Ping from triage @alexreg. Any updates on this? I saw that `miri` `rustfmt` `rust-installer` and `stdsimd` already got merged. Do you want to get this one up to date?

**Comment by alexreg** [comment]
@flip1995 Sorry, completely forget about this! Will get to it tonight.

File: clippy_lints/src/lifetimes.rs [hunk 2]
```diff
@@ -18,7 +18,7 @@ use syntax::symbol::keywords;
 /// them leads to more readable code.
 ///
 /// **Known problems:** Potential false negatives: we bail out if the function
-/// has a `where` clause where lifetimes are mentioned.
+/// has a where-clause where lifetimes are mentioned.
```
[the remark this answers is not in the document]
Comment (alexreg) [reply clippy_lints/src/lifetimes.rs]: Yeah, agreed.

**Comment by alexreg** [comment]
@flip1995 Okay, I've resolved all the points in your review, except `tests/ui/update-all-references.sh` didn't work for me... would you mind blessing the tests, if that works for you? You should be able to push directly to my branch.

Thanks a lot for the review in any case!

**Comment by flip1995** [comment]
Thanks for all the work! Let's see if travis passes and then merge this.

**Comment by alexreg** [comment]
@flip1995 No problem. Appreciate you being open to these. Fingers crossed for Travis...

**Comment by flip1995** [comment]
Some doc tests in clippy_lints are failing. Adding a `ignore` to them should fix this problem.

**Comment by alexreg** [comment]
Thanks!

**Pull request merged**